packages/devtools_app/lib/src/inspector/inspector_screen.dart

```diff
@@ -278,6 +319,40 @@ class InspectorScreenBodyState extends State<InspectorScreenBody>
           //);
         });
       }
+
+      addAutoDisposeListener(
+          inspectorController.selectedNode, _selectedNodeChanged);
+    });
+  }
+
+  void _selectedNodeChanged() {
+    final node = inspectorController.selectedNode.value;
+    final inspectorRef = node?.diagnostic?.valueRef?.id;
+    // Check whether the node that was just selected has any errors associated
+    // with it.
+    var errorIndex = inspectorRef != null
+        ? _errors.keys.toList().indexOf(inspectorRef)
+        : null;
+    if (errorIndex == -1) {
+      errorIndex = null;
+    }
+    // Update the selected index for the error navigator.
+    if (_selectedErrorIndex != errorIndex) {
+      setState(() => _selectedErrorIndex = errorIndex);
+    }
+    // Additionally, mark this error as "read" so that it isn't counted by the badge
+    // (but is still highlighted as an error).
+    if (errorIndex != null) {
+      serviceManager.errorBadgeManager
+          .markErrorAsRead(InspectorScreen.id, _errors[inspectorRef]);
```

Comment:
I think we want to mark all errors as read as soon as we navigate to the inspector screen. Then just depend on the error-navigator UI to guide users to errored widgets.

Reply:
I think that's different to what was noted at https://github.com/flutter/devtools/pull/2630#issuecomment-771168346. It's easy enough to change, but I wanted to double-check!

Reply:
discussed with jacob offline - let's mark errors as read when you visit the screen. I think there could be iterations we could do in a follow up CL to mark the individual error indicators a little differently once each errored widget has been visited, but that can be done in a later CL. Could be good to add a TODO somewhere in the error navigator code to indicate errors as "visited".

Reply:
Done in e5ed5548 and 5a0f9afb. I made the error indicator circles become just an outline once an error is read (rather than removing the "read" code or leaving it unused). I'll post a video of how it currently looks below for review.